frontend/js/src/artist-similarity/src/Data.tsx

```diff
@@ -0,0 +1,168 @@
+    var green = Math.floor(Math.random() * 200);
+    var blue  = Math.floor(Math.random() * 200);
+
+    return Color(red, green, blue);
+}
+
+const mixColor = (color1: ColorType, color2: ColorType, weight: number): ColorType => {
+    var red = Math.floor((color2.red - color1.red) * weight + color1.red);
+    var green = Math.floor((color2.green - color1.green) * weight + color1.green);
+    var blue = Math.floor((color2.blue - color1.blue) * weight + color1.blue);          
+    var opacity = Math.floor((color2.opacity - color1.opacity) * weight + color1.opacity);
+
+    return Color(red, green, blue);
+}
+
+const Data = () => {
+    const LIMIT_VALUE = 18;
+    const BASE_URL = "https://labs.api.listenbrainz.org/similar-artists/json?algorithm=session_based_days_7500_session_300_contribution_5_threshold_10_limit_100_filter_True_skip_30&artist_mbid=";
+    
+    var artist_mbid = "8f6bd1e4-fbe1-4f50-aa9b-94c450ec0f11";
+    var color1 = colorGenerator();
+    var color2 = colorGenerator();
+
+    const [similarArtists, setSimilarArtists] = useState({} as DatasetResponseType);
+    const [artist, setArtist] = useState({} as DatasetResponseType);
+    const [limit, setLimit] = useState(LIMIT_VALUE);
+    const [colors, setColors] = useState([color1, color2]);
+    
+    var transformedArtists = {};
+    var scoreList: Array<number> = [];
+    var similarArtistsList: Array<ArtistType> = [];
+    var mainArtist: ArtistType = {} as ArtistType;
+    var minScore = 0;
+    
+    const fetchData = (artist_mbid: string): void => {
+        fetch(BASE_URL + artist_mbid)
+        .then((response) => response.json())
+        .then((data) => processData(data))   
+    }
+    
+    const processData = (data: Array<ApiResponseType>): void => {
+        let mainArtist = data[1] as DatasetResponseType;
+        setArtist(mainArtist);
+
+        let similarArtists = data[3] as DatasetResponseType;
+        setSimilarArtists(similarArtists);
+        setColors([mixColor(colors[0], colors[1], 0.3), color2]);
+    }
+    
+    useEffect(() => {
+        fetchData(artist_mbid);
+        setColors([color1, color2]);
+    }, []);
+
+    similarArtistsList = similarArtists && similarArtists.data && (similarArtists.data.map((artist: any) => artist));
+    similarArtistsList = similarArtistsList && similarArtistsList.splice(0, limit);
+    
+    mainArtist = artist && artist.data && artist.data[0];
+    similarArtistsList && similarArtistsList.push(mainArtist);
+
+    // Calculating minScore for normalization which is always the last element of the array (because it's sorted)
+    minScore = similarArtistsList && similarArtistsList[LIMIT_VALUE - 1].score as number;
+    minScore = Math.sqrt(minScore);
+
+    transformedArtists = similarArtistsList && {
+        
+        "nodes": similarArtistsList.map((artist: ArtistType, index: number) => {
+
+            if(artist !== mainArtist) {
+                var computedScore = minScore / Math.sqrt(artist.score as number);
```

Comment:
Ah, as a matter of fact I just hit that type of error on your prototype :D
<img width="981" alt="image" src="https://github.com/metabrainz/listenbrainz-server/assets/6179856/7eb29c26-8729-4d8e-8fe0-00fad550f075">

Reply:
I do want to learn a little better how to handle undefined. I tried some changes, basically checking if it's defined and if not setting it to null (i.e. 1 in case of score).